tile-based spectra/redshift column updates

sbailey opened the pull request:
This is the next installment in updates to the data models in DESI_SPECTRO_REDUX/SPECPROD/tiles/GROUPTYPE/TILEID/GROUPID, with columns descriptions updated with
```
update_column_descriptions -i filename.rst [--inplace --force]
```
I first ran without `--inplace --force` to see what it would update, and then updated `py/desidatamodel/data/column_descriptions.csv` if I thought any pre-existing description was better than what we had in column_descriptions.csv.  I then reran with `--inplace --force` to do the actual updates.  Exception: I kept the zmtl ZWARN and ZTILED slightly different from the standard descriptions, since zmtl augments the ZWARN flags from redrock, and the standard ZTILEID description is for the mtl ledgers where there actually can be more than one tile, which doesn't apply to zmtl.

This PR also includes the new option `--inplace`, which is a convenience option equivalent to providing the same `--outfile` as `--infile` (i.e. so you only have to write the input filename once, and don't risk a cut-and-paste error or replacing one datamodel with another).

weaverba137 commented on doc/DESI_SPECTRO_REDUX/SPECPROD/tiles/GROUPTYPE/TILEID/GROUPID/qso_qn-SPECTROGRAPH-TILEID-GROUPID.rst:
```diff
@@ -58,40 +58,41 @@ Required Data Table Columns
+Notes:
 
-.. [3] The QN selection is performed with these parameters. As it stands, in QN afterburner everything with np.max(confindence) > 0.5 is considered as a quasar. However, specific cut will be used depends on each target class; QSO_target will use np.max(confidence) > 0.95.
+ * Z_QN is the redshift estimated on the line of the highest confidence
```
Be careful here: you've got some indentation on these bullet points, which renders as a blockquote containing a bullet list rather than just a bullet list.  The indentation on line 96 may also be a problem.

weaverba137 reviewed (approved):
Just a minor recommendation to clean up formatting on one file where you replaced some footnotes with a bullet list.

weaverba137 commented:
Also, please update the change log. You may want to merge or rebase with `main` to avoid possible conflicts on that file.

weaverba137 commented:
@sbailey, in order to move this along, do you want me to make the requested changes myself?

sbailey commented:
@weaverba137 thanks for the offer.  I updated the change log and fixed the formatting; the delay was due to also getting clarification about the meaning of some of the columns and footnotes, which I have now also updated.  If you can take it from here, I would appreciate that.  Thanks.

weaverba137 commented:
OK, sounds good.
